commit 5311fec82d87baf90b379b5c1369661f013265be
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Wed Mar 1 16:06:53 2017 +0000

    clk-bcm2835: Correct the prediv logic
    
    If a clock has the prediv flag set, both the integer and fractional
    parts must be scaled when calculating the resulting frequency.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/clk/bcm/clk-bcm2835.c b/drivers/clk/bcm/clk-bcm2835.c
index eec6735505c0..e0d28add45ef 100644
--- a/drivers/clk/bcm/clk-bcm2835.c
+++ b/drivers/clk/bcm/clk-bcm2835.c
@@ -616,8 +616,10 @@ static unsigned long bcm2835_pll_get_rate(struct clk_hw *hw,
 	using_prediv = cprman_read(cprman, data->ana_reg_base + 4) &
 		data->ana->fb_prediv_mask;
 
-	if (using_prediv)
+	if (using_prediv) {
 		ndiv *= 2;
+		fdiv *= 2;
+	}
 
 	return bcm2835_pll_rate_from_divisors(parent_rate, ndiv, fdiv, pdiv);
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1029_5311fec82d87baf90b379b5c1369661f013265be.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1029_5311fec82d87baf90b379b5c1369661f013265be.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
